Python/chapter02/2.6 - Palindrome/miguel_2.6_soln.py

```diff
@@ -154,6 +155,29 @@ def reverse_linked_list(ll: LinkedList) -> LinkedList:
     return output_ll
 
 
+def palindrome_constant_space(ll: LinkedList) -> bool:
+    """
+    Same as the palindrome function below, but
+    with constant space
+    Runtime:  O(n^2)
+    Space Complexity:  O(1)
+    :param ll: an input linked list
+    :return: true if ll is a palindrome, false otherwise
+    """
+    n1 = ll.head
+    for i in range(math.ceil(ll.size/2)):
+        curr_data = n1.data
+        n2 = n1
+        # advance n2 pointer to desired complement index
+        for j in range(i, ll.size - i - 1):
+            n2 = n2.next
+        # if not the same, then not a palindrome
+        if n2.data != curr_data:
+            return False
+        n1 = n1.next
+    return True
+
+
 def palindrome(ll: LinkedList) -> bool:
```

Comment:
forgot this one in the previous iteration: nit: call this `is_palindrome` since it returns a boolean. that way you can more easily guess what itś going to do.

Reply:
updated